sds_data_manager/lambda_code/SDSCode/download_api.py

```diff
@@ -1,30 +1,104 @@
-    """This allows user to download file from S3 using pre-signed URL generated
-    by the download query API.
+def http_response(headers=None, status_code=200, body="Success"):
+    """Customizes HTTP response for the lambda function.
+
+    Parameters
+    ----------
+    headers : dict, optional
+        Content headers for the response, defaults to Content-type: text/html.
+    status_code : int, optional
+        HTTP status code indicating the result of the operation, defaults to 200.
+    body : str, optional
+        The content of the response, defaults to 'Success'.
 
-    Args:
-        filename_and_path (str): exact path with filename where user want to store.
-            Eg. dir/subdir/filename.ext
-        download_link (str): pre-signed URL from S3
+    Returns
+    -------
+    dict
+        A dictionary containing headers, status code, and body, designed to be returned
+        by a Lambda function as an API response.
     """
-    # Get file content using urlopen
-    with urlopen(download_link) as response:
-        if response.getcode() != 200:
-            logger.warn(
-                "Failed to download file [%s], returned status code [%d]",
-                download_link,
-                response.status_code,
-            )
-            return
-        # save/write file content to file on local machine
-        with open(filename_and_path, "wb") as file:
-            logger.info(f"Downloading to {filename_and_path}")
-            file.write(response.read())
+    if headers is None:
+        headers = (
+            {
+                "Content-Type": "text/html",
+            },
+        )
+    return {
+        "headers": headers,
+        "statusCode": status_code,
+        "body": body,
+    }
+
+
+def lambda_handler(event, context):
+    """This lambda handler checks if this file exists or not. If file doesn't exist, it
+    gives back an error. Otherwise, it returns pre-signed s3 url that user can use to
+    download data from s3.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    context : LambdaContext
+        This object provides methods and properties that provide information
+        about the invocation, function, and runtime environment.
+
+    Returns
+    -------
+    dict
+        The response from the function which could either be a pre-signed
+        S3 URL in case of successful operation or an error message with
+        corresponding status code in case of failure.
+    """
+    one_day = 86400
+    url_life = os.environ.get("URL_EXPIRE", one_day)
+    path_params = event.get("pathParameters", {}).get("proxy", None)
+    logger.debug("Parsing path parameters=[%s] from event=" "[%s]", path_params, event)
+    if not path_params:
+        response_body = (
+            "No file requested for download. Please provide a filename "
+            "in the path. Eg. /download/filename.pkts"
```

Comment:
Should we add this so that user know that we expect full path on s3 since that's how we check in the try/except.
```suggestion
            "in the path. Eg. /download/<filepath-if-exist>/filename.pkts"
```

Reply:
Good point, I'll update this!